### PR TITLE
verify all leg data is coming through TM-4029

### PR DESCRIPTION
**Pull request by SidelineCory24**
**Note: most of these to-do's will apply to the following pages: Agenda Item History, Agenda Item Maintenance, Panel Meeting Agenda**

To-do:
- [x] Show `None listed` if instead of a blank value
- [x] Remove icons for ORG, ETA, & TED
- [x] Verify all leg data is being ingested
- [x] Insert legacy value into dropdown for AIM

Verify the following is disabled
- [x] Save button
- [x] Employee, Asg, Sep, Bids Dropdown
- [x] Status Dropdown
- [x] Add position number
- [x] Report Category Dropdown
- [x] Panel Dates Dropdown (ID & ML)
- [x] Ability to remove remarks from AIM Pane
- [x] Ability to remove legs from AI Timeline
- [x] TED calendar
- [x] TOD dropdown
- [x] Action dropdown
- [x] Travel dropdown
- [x] Ability to add/remove from Frequent Positions
- [x] Ability to add/remove from Remarks Glossary

### CODE REVIEW

File: src/Components/Agenda/AgendaItemTimeline/AgendaItemTimeline.jsx [hunk 1]
```diff
@@ -8,12 +8,144 @@ import { EMPTY_FUNCTION } from 'Constants/PropTypes';
...
 
 const AgendaItemTimeline = ({ unitedLoading, setParentLoadingState, updateLegs,
+  // eslint-disable-next-line no-unused-vars
   asgSepBid, efPos, agendaItemLegs }) => {
   const pos_results = useSelector(state => state.positions);
   const pos_results_loading = useSelector(state => state.positionsIsLoading);
   const pos_results_errored = useSelector(state => state.positionsHasErrored);
 
-  const [legs, setLegs] = useState(agendaItemLegs);
+  const legs$$ = [
```
Comment (SidelineCory24) [inline src/Components/Agenda/AgendaItemTimeline/AgendaItemTimeline.jsx:17]: this is for dummy data purposes to see that the legacy value (value not in our dropdown) will show in the dropdown, will revert before merging

File: src/Components/Agenda/AgendaItemLegsForm/AgendaItemLegsForm.jsx [hunk 2]
```diff
@@ -62,6 +62,48 @@ const AgendaItemLegsForm = props => {
     'Travel',
   ];
 
+  useEffect(() => {
+    legs.forEach(l => {
+      const isLegacyValue = (!includes(TODs, l.tourOfDutyCode) && l.tourOfDutyCode !== '');
+      if (isLegacyValue) {
+        TODs.push({
+          id: 'LEGACY TOD',
+          code: 'LEGACY TOD',
```
Comment (SidelineCory24) [inline src/Components/Agenda/AgendaItemLegsForm/AgendaItemLegsForm.jsx:71]: will verify in dev1 what data is coming through for legacy values besides the text

**Comment by SidelineCory24** [comment]
adding legacy value to dropdown (value that is not in the dropdown)
image

image

image

File: src/Components/Agenda/AgendaItemMaintenanceContainer/AgendaItemMaintenanceContainer.jsx [hunk 3]
```diff
@@ -25,6 +26,8 @@ const AgendaItemMaintenanceContainer = (props) => {
   const agendaID = get(props, 'match.params.agendaID') || '';
   const { data: agendaItemData, error: agendaItemError, loading: agendaItemLoading } = useDataLoader(api().get, `/fsbid/agenda/agenda_items/${agendaID}/`);
   const agendaItem = get(agendaItemData, 'data') || {};
+  const isReadOnly = true;
```
Comment (SidelineCory24) [inline src/Components/Agenda/AgendaItemMaintenanceContainer/AgendaItemMaintenanceContainer.jsx:29]: setting to true to be able to test disabled view with legacy dropdown values, remove this hard coding and the hard coded legs$$ for the legacy values to test as well.

File: src/Components/Agenda/AgendaItemTimeline/AgendaItemTimeline.jsx [hunk 4]
```diff
@@ -8,12 +8,48 @@ import { EMPTY_FUNCTION } from 'Constants/PropTypes';
...
 const AgendaItemTimeline = ({ unitedLoading, setParentLoadingState, updateLegs,
-  asgSepBid, efPos, agendaItemLegs }) => {
+  // eslint-disable-next-line no-unused-vars
+  asgSepBid, efPos, agendaItemLegs, isReadOnly }) => {
   const pos_results = useSelector(state => state.positions);
   const pos_results_loading = useSelector(state => state.positionsIsLoading);
   const pos_results_errored = useSelector(state => state.positionsHasErrored);
 
-  const [legs, setLegs] = useState(agendaItemLegs);
+  const legacyLeg = [
```
Comment (SidelineCory24) [inline src/Components/Agenda/AgendaItemTimeline/AgendaItemTimeline.jsx:17]: hard coded to test legacy leg data for dropdown, remove and comment in line 51 to test normally

will remove before merging

File: src/sass/_agendaItemHistory.scss [hunk 5]
```diff
@@ -287,7 +287,7 @@
 
           th {
             border: none;
-            padding: .5rem 1rem;
+            padding: .8rem 1rem;
```
Comment (elizabeth-jimenez) [inline src/sass/_agendaItemHistory.scss:290]: Let's stay away from relative units

Comment (SidelineCory24) [reply src/sass/_agendaItemHistory.scss]: done

File: src/sass/_agendaItemLegsForm.scss [hunk 6]
```diff
@@ -70,6 +70,16 @@
...
+  .icon-disabled {
+    color: $color-gray;
+    .fa {
+      cursor: not-allowed;
+    }
+  }
+  .icon-disabled:hover {
+    color: $color-gray;
+    cursor: not-allowed;
+  }
```
Comment (scott062) [inline src/sass/_agendaItemLegsForm.scss:82]: Consolidate disabled stylings to parent class at uppermost parent level where applicable

File: src/Components/Agenda/AgendaItemResearchPane/AgendaItemResearchPane.jsx [hunk 7]
```diff
@@ -121,6 +121,7 @@ const AgendaItemResearchPane = forwardRef((props = { perdet: '', clientData: {},
               positions={frequentPositions}
               addFrequentPosition={addFrequentPosition}
```
Comment (scott062) [inline src/Components/Agenda/AgendaItemResearchPane/AgendaItemResearchPane.jsx:122]: isReadOnly ? () => {} : addFrequentPosition

File: src/sass/_panelMeetingAgendas.scss [hunk 8]
```diff
@@ -164,7 +164,7 @@
 
           th {
             border: none;
-            padding: .5rem 1rem;
+            padding: .8rem 1rem;
```
Comment (scott062) [inline src/sass/_panelMeetingAgendas.scss:167]: Same as Elizabeths comment

File: src/sass/_agendaItemLegsForm.scss (hunk 6 above)
Comment (SidelineCory24) [reply src/sass/_agendaItemLegsForm.scss:82]: discussed offline, consolidated

File: src/Components/Agenda/AgendaItemResearchPane/AgendaItemResearchPane.jsx (hunk 7 above)
Comment (SidelineCory24) [reply src/Components/Agenda/AgendaItemResearchPane/AgendaItemResearchPane.jsx]: done for fp, ai timeline, and remarks gloss

File: src/sass/_panelMeetingAgendas.scss (hunk 8 above)
Comment (SidelineCory24) [reply src/sass/_panelMeetingAgendas.scss]: done

**Review by scott062 — APPROVED** [review]

**Pull request merged**